ts/App.tsx

```diff
@@ -33,13 +34,15 @@ configurePushNotifications(store);
 export default class App extends React.Component<{}, never> {
   public render() {
     return (
-      <StyleProvider style={theme()}>
-        <Provider store={store}>
-          <PersistGate loading={undefined} persistor={persistor}>
-            <RootContainer />
-          </PersistGate>
-        </Provider>
-      </StyleProvider>
+      <MenuProvider>
```

Comment:
why at the topmost level and not at the inner level? what's the difference / what are the implications (if any)? - cc @fpersico

Reply:
could it make sense to just wrap the screens that need this functionality instead of the whole app?

Reply:
@cloudify 
Usually Providers uses Context to share information with the child components. In this case i don't think it make any difference but probably is safer to add the new Provider after the PersistGate just before the RootContainer.